### PR TITLE
Upgrade to hapi 17

### PR DESCRIPTION
**Pull request by deeptibaghel**

### CODE REVIEW

**Comment by vladikoff** [comment]
@vbudhram r? so we can use it with hapi 17 in auth and oauth

File: tests/index.js [hunk 1]
```diff
@@ -80,28 +78,29 @@ var passingTestCases = [
...
+    })
+    it(testCase.name, () => {
+      server.inject(requestOptions, function (response) {
+      assert.equal(response.headers[testCase.expectedKey], testCase.expectedHeader)
+     })
+    })
+  })
 })
 
+
```
Comment (vbudhram) [inline tests/index.js]: Could remove extra line break.

**Review by vbudhram — CHANGES_REQUESTED** [review]
@deeptibaghel Looks good! Could you update the README example, something along these lines

```javascript
  try{
   await server.register({
      plugin: require('../index.js'),
      options: hpkpOptions
    })
  } catch(err) {
    console.error('Failed to load plugin:', err)
}
```

File: README.md [hunk 2]
```diff
@@ -18,12 +18,14 @@ var hpkpOptions = {
...
-  options: hpkpOptions
-}, function (err) {
-  if (err) {
-    console.error('Failed to load plugin:', err)
-  }
-})
+async () => {
+    try {
+        await server.register({
+          plugin: require('../index.js'),
```
Comment (vbudhram) [inline README.md]: For this example, I think this should be `plugin: hpkp`.

**Review by vbudhram — COMMENTED** [review]
@deeptibaghel Just the one thing and this should be good. Once merged, I can tag a new release.

**Comment by vbudhram** [comment]
@deeptibaghel Nice work!

**Pull request merged**